### PR TITLE
Fix #9551 - Update LCC using NIST 135 supplement for years 2020, 2021 and 2022

### LINKED ISSUE #9551: Update LCC using NIST 135
Issue overview
--------------
The last update for the energy cost price escalation was LCCusePriceEscalationDataSet2019.idf. The 2020, 2021, and 2022 updates should be added based on NIST 135 annual supplements. See #7218 for the last update

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [x] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

 - Fix #9551 
 - I added (and commited) a python script with no dependencies to process the encostXX.txt files into a formatted IDF.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [x] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [x] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [x] If structural output changes, add to output rules file and add OutputChange label
 - [x] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: cmake/Install.cmake [hunk 1]
```diff
@@ -286,6 +286,9 @@ install(FILES "${PROJECT_SOURCE_DIR}/datasets/LCCusePriceEscalationDataSet2016.i
 install(FILES "${PROJECT_SOURCE_DIR}/datasets/LCCusePriceEscalationDataSet2017.idf" DESTINATION "./DataSets" COMPONENT Datasets)
 install(FILES "${PROJECT_SOURCE_DIR}/datasets/LCCusePriceEscalationDataSet2018.idf" DESTINATION "./DataSets" COMPONENT Datasets)
 install(FILES "${PROJECT_SOURCE_DIR}/datasets/LCCusePriceEscalationDataSet2019.idf" DESTINATION "./DataSets" COMPONENT Datasets)
+install(FILES "${PROJECT_SOURCE_DIR}/datasets/LCCusePriceEscalationDataSet2020.idf" DESTINATION "./DataSets" COMPONENT Datasets)
+install(FILES "${PROJECT_SOURCE_DIR}/datasets/LCCusePriceEscalationDataSet2021.idf" DESTINATION "./DataSets" COMPONENT Datasets)
+install(FILES "${PROJECT_SOURCE_DIR}/datasets/LCCusePriceEscalationDataSet2022.idf" DESTINATION "./DataSets" COMPONENT Datasets)
```
Comment (jmarrec) [inline cmake/Install.cmake:291]: Install new datasets

File: datasets/LCCusePriceEscalationDataSet2019.idf [hunk 2]
```diff
@@ -1,2606 +1,2539 @@
...
+! https://www.energy.gov/sites/prod/files/2019/05/f62/encost19.txt
+
+LifeCycleCost:UsePriceEscalation,
+    NorthEast Residential-Electricity,  !- Name
+    Electricity,             !- Resource
+    2020,                    !- Escalation Start Year
+    January,                 !- Escalation Start Month
+    1.0138,                  !- Year 1  Escalation [2020]
+    1.0239,                  !- Year 2  Escalation [2021]
+    1.0336,                  !- Year 3  Escalation [2022]
```
Comment (jmarrec) [inline datasets/LCCusePriceEscalationDataSet2019.idf:27]: I redid the 2019 one with the script, to ensure it wasn't producing any actual diffs (it isn't)

File: scripts/dev/update_nist_escalation.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+FUEL_NAME_CONVERSIONS = {
+    # NIST 135 Name: EnergyPlus Name
+    "electricity": "Electricity",
+    "distillate oil": "FuelOilNo1",
+    "natural gas": "NaturalGas",
+    "lpg": "Propane",
+    "residual oil": "FuelOilNo2",
+    "coal": "Coal",
+}
```
Comment (jmarrec) [inline scripts/dev/update_nist_escalation.py:84]: Helper script to parse the encostXX.txt and produce a nicely formatted IDF

File: datasets/LCCusePriceEscalationDataSet2019.idf (hunk 2 above)
Comment (jmarrec) [reply datasets/LCCusePriceEscalationDataSet2019.idf:27]: the format is slightly different (and field names are more explicit) but that's it.

**Comment by Myoldmopar** [comment]
@jmarrec there's a license text test failure here.  I haven't reviewed it yet, but the tests seem all clean other than that.  I can plan on just updating the license text locally while testing and pushing the changes right before merge to avoid wasting CI runs.

File: cmake/Install.cmake (hunk 1 above)
Comment (Myoldmopar) [inline cmake/Install.cmake:291]: Yeah good catch.

**Review by Myoldmopar — COMMENTED** [review]
This looks good, although I don't know that we have any automated tests covering these files.  I'll have to check.

**Comment by Myoldmopar** [comment]
Yep, we do here: https://github.com/NREL/EnergyPlus/blob/develop/tst/EnergyPlus/unit/Datasets.unit.cc#L163

I'm going to fix the license, and add those lines for the new yearly data files you added here, and let CI re-run overnight.  If all clean, this can surely merge in the morning.

**Comment by Myoldmopar** [comment]
Alright pulled develop in, fixed the license text, and added unit test support for the new files.  The new unit tests, along with the full test suite, pass just fine.  I'll give the GHA CI workflows a minute to grab the repo before merging and deleting the branch, but this is ready to merge.  Thanks @jmarrec

**Comment by jmarrec** [comment]
@Myoldmopar Thanks for the review and the fixes!

**Comment by JasonGlazer** [comment]
Thanks @jmarrec !

**Pull request merged**